### ISSUE TITLE
RF Sync header being sent at the end of the code payload instead of at the beggining 

### ISSUE THREAD

**Issue by Rodrigoah** [esphome/issues#1843]
**Operating environment/Installation (Hass.io/Docker/pip/etc.):**

Hass.io

**ESP (ESP32/ESP8266, Board/Sonoff):**

NodeMcu

**ESPHome version (latest production, beta, dev branch)**

1.15.3

**Affected component:**

remote_transmitter

**Description of problem:**

Header of custom protocol is sent at the end of the code instead of at the beggining

**Problem-relevant YAML-configuration entries:**
```yaml
      remote_transmitter.transmit_rc_switch_raw:
        code: '1010001100110000000101000010010000000001000000000000101101110100'
        protocol:
          pulse_length: 510
          sync: [25, 3]
          zero: [1,3]
          one: [3,1]
```

**Logs (if applicable):**

```
PASTE DEBUG LOG HERE

```

**Additional information and things you've tried:**

**Comment by Rodrigoah** [comment]
![image](https://user-images.githubusercontent.com/10799831/107985223-f8e81100-6fc9-11eb-8468-71c076e5201b.png)

As you can see, the sync is being sent at the end, while it is starting with the code

**Comment by Rodrigoah** [comment]
more info, 

it does not happen if "times" is set to a value greater than one

**Comment by glmnet** [comment]
Can you update with 1.16.2?

**Comment by Rodrigoah** [comment]
> Can you update with 1.16.2?

This issue is solved in 1.16.2

The issue can be closed
